Source/WebCore/Modules/webxr/WebXROpaqueFramebufferCocoa.cpp

```diff
@@ -370,31 +349,40 @@ static PlatformXR::Layout displayLayout(const PlatformXR::FrameData::LayerSetupD
     return data.physicalSize[1][0] > 0 ? PlatformXR::Layout::Layered : PlatformXR::Layout::Shared;
 }
 
+static IntSize calcFramebufferPhysicalSize(const IntSize& leftPhysicalSize, const IntSize& rightPhysicalSize)
+{
+    if (rightPhysicalSize.isEmpty())
+        return leftPhysicalSize;
+    RELEASE_ASSERT(leftPhysicalSize.height() == rightPhysicalSize.height(), "Only side-by-side shared framebuffer layout is supported");
+    return { leftPhysicalSize.width() + rightPhysicalSize.width(), leftPhysicalSize.height() };
+}
+
 bool WebXROpaqueFramebuffer::setupFramebuffer(GraphicsContextGL& gl, const PlatformXR::FrameData::LayerSetupData& data)
 {
-    auto framebufferSize = IntSize(data.framebufferSize[0], data.framebufferSize[1]);
+    auto leftPhysicalSize = toIntSize(data.physicalSize[0]);
+    auto rightPhysicalSize = toIntSize(data.physicalSize[1]);
+    auto framebufferSize = calcFramebufferPhysicalSize(leftPhysicalSize, rightPhysicalSize);
     bool framebufferResize = !m_drawAttachments || m_framebufferSize != framebufferSize || m_displayLayout != displayLayout(data);
-    bool foveationChange = !data.horizontalSamples[0].empty() && !data.verticalSamples.empty() && !data.horizontalSamples[1].empty();
-    m_usingFoveation = foveationChange;
+    bool foveationChange = m_usingFoveation ^ data.foveationRateMapDesc.screenSize.isEmpty();
```

Comment:
Did you mean:
```
bool foveationChange = m_usingFoveation ^ !data.foveationRateMapDesc.screenSize.isEmpty();
```

Maybe this can be clearer as:
```
    bool usingFoveation = !data.foveationRateMapDesc.screenSize.isEmpty();
    bool foveationChange = m_usingFoveation ^ usingFoveation;
```
and later we can do:
```
m_usingFoveation = usingFoveation;
```